smartsim/_core/entrypoints/dragon_client.py

```diff
@@ -0,0 +1,107 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterpris
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import json
+import os
+import signal
+import sys
+import time
+import typing as t
+
+import zmq
+
+from smartsim._core.launcher.dragon.dragonConnector import DragonConnector
+from smartsim._core.schemas import (
+    DragonHandshakeRequest,
+    DragonRequest,
+    DragonShutdownRequest,
+    request_registry,
+)
+from smartsim.log import get_logger
+
+SIGNALS = [signal.SIGINT, signal.SIGQUIT, signal.SIGTERM, signal.SIGABRT]
+
+logger = get_logger("Dragon Client")
+
+
+def cleanup() -> None:
+    logger.debug("Cleaning up")
+
+
+def main(args: argparse.Namespace) -> int:
+
+    requests: t.List[DragonRequest] = []
+
+    with open(args.submit, "r", encoding="utf-8") as request_file:
```

Comment:
we might want to gate this to give a nicer error message if the file is missing. 

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'bar'
>>> import json
>>> 
>>> 
>>> with open(foo, "r", encoding="utf-8") as rf:
...   req_strings = json.load(fp=rf)
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'bar'
```

maybe as simple as:

`if pathlib.Path(foo).exists(): ...`

Reply:
Malformed json would also lead to a failure that we don't handle

Reply:
Both are catastrophic and unrecoverable, what about raising the error adding a more informative message to it?